llvm/include/llvm/MC/MCSymbolRepo.h

```diff
@@ -23,6 +23,21 @@ class MCSymbolRepo : public MCSymbol {
 
   // A pointer to the TiketNode metadata of the corresponding GlobalObject.
   const TicketNode *CorrespondingTicketNode = nullptr;
+
+  // Get the symbol full name.
+  static const std::string getFullName(const MCContext &Ctx,
+                                       const StringRef InitialName,
+                                       const ticketmd::DigestType &Digest) {
+    if (Digest != ticketmd::NullDigest) {
+      return (Twine(Ctx.getAsmInfo()->getPrivateGlobalPrefix()) +
+              Twine(InitialName) + Twine(".") + Twine(Digest.digest()))
+          .str();
+    } else {
+      return (Twine(Ctx.getAsmInfo()->getPrivateGlobalPrefix()) +
+              Twine(InitialName))
+          .str();
```

Comment:
There's a fair chunk of duplicated code here (which wasn't duplicated in the previous version). Could we go back to that previous style (but keep the use of Twine)?

Reply:
Thanks Paul, fixed it.